pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5669,106 +5742,89 @@ do_file_write_operation(struct pt_regs *ctx, u32 event_id, u32 tail_call_id)
         return 0;
     }
 
-    if (!should_submit(VFS_WRITE, config) && !should_submit(VFS_WRITEV, config) &&
-        !should_submit(__KERNEL_WRITE, config) && !should_submit(MAGIC_WRITE, config)) {
+    if (!should_submit_io_event(config, event_id) && !should_submit(MAGIC_WRITE, config)) {
         bpf_tail_call(ctx, &prog_array, tail_call_id);
         del_args(event_id);
         return 0;
     }
 
     loff_t start_pos;
-    void *ptr;
-    struct iovec *vec;
-    size_t count;
-    unsigned long vlen;
+    io_data_t io_data;
+    file_info_t file_info;
 
     struct file *file = (struct file *) saved_args.args[0];
-    void *file_path = get_path_str(GET_FIELD_ADDR(file->f_path));
+    file_info.pathname_p = get_path_str(GET_FIELD_ADDR(file->f_path));
 
-    if (event_id == VFS_WRITE || event_id == __KERNEL_WRITE) {
-        ptr = (void *) saved_args.args[1];
-        count = (size_t) saved_args.args[2];
+    io_data.is_buf = is_buf;
+    if (is_buf) {
+        io_data.buf.ptr = (void *) saved_args.args[1];
+        io_data.buf.len = (size_t) saved_args.args[2];
     } else {
-        vec = (struct iovec *) saved_args.args[1];
-        vlen = saved_args.args[2];
+        io_data.vec.ptr = (struct iovec *) saved_args.args[1];
+        io_data.vec.vlen = saved_args.args[2];
     }
     loff_t *pos = (loff_t *) saved_args.args[3];
 
     // Extract device id, inode number, and pos (offset)
-    dev_t s_dev = get_dev_from_file(file);
-    unsigned long inode_nr = get_inode_nr_from_file(file);
+    file_info.device = get_dev_from_file(file);
+    file_info.inode = get_inode_nr_from_file(file);
     bpf_probe_read(&start_pos, sizeof(off_t), pos);
 
     bool char_dev = (start_pos == 0);
-    u32 bytes_written = PT_REGS_RC(ctx);
-    u32 header_bytes = FILE_MAGIC_HDR_SIZE;
-    if (header_bytes > bytes_written)
-        header_bytes = bytes_written;
+    u32 bytes_read = PT_REGS_RC(ctx);
```

Comment:
this could also be bytes_written, no? if so, should we rename it to bytes_changed, or something else ?